snakefmt/formatter.py

```diff
@@ -153,8 +153,12 @@ def run_black_format_str(
                 tmpstring += line
             string = textwrap.dedent(tmpstring)
 
+        needs_artifical_nest = inside_nested_statement and not string.startswith("if")
+        if needs_artifical_nest:
+            string = f"if x:\n{textwrap.indent(string, TAB)}"
+
```

Comment:
I see, here you make sure `black` will indent python code with one newline only if we're in a nested context. Nice.

Reply:
I looked into the `black` `Mode` class that we tap into to modify line length, in case there was a way to tell black we're in a nested context - no such. I think your fix is good enough in that respect

Reply:
Yeah, it feels so hacky, but as you say, there's no other way of passing that indentation context to black (which makes sense I guess).